hil/ext/switches/dellnos9.py

```diff
@@ -0,0 +1,377 @@
+            interface.replace('/', '-')
+        payload = '<interface><name>%s</name><portmode><hybrid>false' \
+                  '</hybrid></portmode><shutdown>true</shutdown>' \
+                  '</interface>' % interface
+
+        self._make_request('PUT', url, data=payload)
+
+    def _port_on(self, interface):
+        """ Turns on <interface>
+
+        Turn on port and enable hybrid portmode and switchport.
+        """
+
+        url = self._construct_url(interface=interface)
+        interface = self._convert_interface(self.interface_type) + \
+            interface.replace('/', '-')
+        payload = '<interface><name>%s</name><portmode><hybrid>true' \
+                  '</hybrid></portmode><switchport></switchport>' \
+                  '<shutdown>false</shutdown></interface>' % interface
+
+        self._make_request('PUT', url, data=payload)
+
+    def _is_port_on(self, port):
+        """ Returns a boolean that tells the status of a switchport"""
+
+        # the url here requires a suffix to GET the shutdown tag in response.
+        url = self._construct_url(interface=port) + '\?with-defaults'
+        response = self._make_request('GET', url)
+        root = etree.fromstring(response.text)
+        shutdown = root.find(self._construct_tag('shutdown')).text
+        # is error handling needed here?
+        return (shutdown == 'false')
+
+    # HELPER METHODS *********************************************
+
+    def _execute(self, interface, command_type, command):
+        """This method gets the url & the payload and executes <command>"""
+        url = self._construct_url()
+        payload = self._make_payload(command_type, command)
+        return self._make_request('POST', url, data=payload)
+
+    def _construct_url(self, interface=None):
+        """ Construct the API url for a specific interface.
+
+        Args:
+            interface: interface to construct the url for
+
+        Returns: string with the url for a specific interface and operation
+
+        If interface is None, then it returns the URL for REST API CLI.
+        """
+
+        if interface is None:
+            return '%s/api/running/dell/_operations/cli' % self.hostname
+
+        val = re.compile(r'^\d+/\d+(/\d+)?$')
+        if val.match(interface):
+            # if `interface` refers to port name
+            # the urls have dashes instead of slashes in interface names
+            interface = interface.replace('/', '-')
+            interface_type = self._convert_interface(self.interface_type)
+        else:
+            # interface refers to `vlan`
+            interface_type = 'vlan-'
+
+        return ''.join([self.hostname, '/api/running/dell/interfaces/'
+                       'interface/', interface_type, interface])
+
+    @staticmethod
+    def _convert_interface(interface_type):
+        """ Convert the interface name from switch CLI form to what the API
+        server understands.
```

Comment:
This docstring doesn't look like it's quite right -- you're converting interface *types*, not full names of specific interfaces, unless I'm mistaken.